CMakeLists.txt

```diff
@@ -1133,7 +1133,14 @@ macro(DR_target_install_dst dst)
     else ()
       set(inst_dst ${dst})
     endif ()
-    install(${ARGN} DESTINATION ${inst_dst})
+    set(export_flag "")
+    foreach (n ${ARGN})
```

Comment:
In case you're wondering why this strange loop: I have tried to get the 'optional' arguments with list(GET ARGN..) etc., but nothing like this seems to work deterministically. The problem seems to be that when function A calls function B, in cmake ARGC is not always == list(LENGTH ARGV) etc., same for ARGN.

Reply:
You didn't like this loop, but I don't see a reply: the examples for how to turn ARGN into a cmake var aren't any better?

Reply:
I spent an hour trying around to make it work w/o the loop, but the list(LENGTH) of ARGN, or any way trying to index into ARGN doesn't work. Browsing online when having nested function calls this is known in cmake. For example, you get list(LENGTH ARGN var) var == 0, even though iterating through ARGN gives you the correct vars. I had found no way to work around (e.g. assigning to a list, constructing a list from the string etc., passing it per string) In contrast, the way above worked reliably

Reply:
I see. Ok, good point. It may work assigning set(mylist "${ARGN}") and then indexing into mylist, though I thought I tried exactly that and there was a problem. I also think above is ok, it works and is already tested, so I'll leave it.